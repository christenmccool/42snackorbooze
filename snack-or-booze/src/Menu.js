import React from "react";
import { Link } from "react-router-dom";
import "./Menu.css";
import {
  Card,
  CardBody,
  CardTitle,
  CardText,
  ListGroup,
  ListGroupItem
} from "reactstrap";

/* Display snacks or drinks menu, linking to each item */
function Menu({ items, type, title, description }) {
  return (
    <section className="col-md-4">
      <Card>
        <CardBody>
          <CardTitle className="font-weight-bold text-center">
            {title}
          </CardTitle>
          <CardText className="font-italic">
            {description}
          </CardText>
          <ListGroup>
            {items.map(item => (
              <Link to={`/${type}/${item.id}`} key={item.id}>
                <ListGroupItem>{item.name}</ListGroupItem>
              </Link>
            ))}
          </ListGroup>
          <Link exact to={`/${type}/add`} className="add-link">Add a new item</Link>
        </CardBody>
      </Card>
    </section>
  );
}

export default Menu;
