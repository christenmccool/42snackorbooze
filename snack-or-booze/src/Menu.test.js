import { render, asFragment, fireEvent, waitFor } from '@testing-library/react';
import Menu from './Menu';
import { MemoryRouter } from 'react-router-dom';
import {snacks} from './mockTestDb';
import '@testing-library/jest-dom/extend-expect';
import '@testing-library/jest-dom';


test('renders without crashing', () => {
    render(<MemoryRouter><Menu items={snacks} type='snacks' title='Snack Menu' description='Tasty Snacks' /></MemoryRouter>);
});

test("if matches snapshot", () => {
    const {asFragment} = render(<MemoryRouter><Menu items={snacks} type='snacks' title='Snack Menu' description='Tasty Snacks' /></MemoryRouter>);
    expect(asFragment()).toMatchSnapshot();
})

test("if NavBar renders", async () => {
    const { getByText, debug } = render(<MemoryRouter><Menu items={snacks} type='snacks' title='Snack Menu' description='Tasty Snacks' /></MemoryRouter>);

    expect(getByText(`Snack Menu`)).toBeInTheDocument();
    expect(getByText(`Tasty Snacks`)).toBeInTheDocument();
    expect(getByText(`Nachos`)).toBeInTheDocument();
    expect(getByText(`Hummus`)).toBeInTheDocument();
    expect(getByText(`Arugula and Walnut Salad`)).toBeInTheDocument();

})