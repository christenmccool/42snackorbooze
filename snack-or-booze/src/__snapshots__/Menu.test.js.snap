// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`if matches snapshot 1`] = `
<DocumentFragment>
  <section
    class="col-md-4"
  >
    <div
      class="card"
    >
      <div
        class="card-body"
      >
        <h5
          class="font-weight-bold text-center card-title"
        >
          Snack Menu
        </h5>
        <p
          class="font-italic card-text"
        >
          Tasty Snacks
        </p>
        <ul
          class="list-group"
        >
          <a
            href="/snacks/nachos"
          >
            <li
              class="list-group-item"
            >
              Nachos
            </li>
          </a>
          <a
            href="/snacks/hummus"
          >
            <li
              class="list-group-item"
            >
              Hummus
            </li>
          </a>
          <a
            href="/snacks/arugula-and-walnut-salad"
          >
            <li
              class="list-group-item"
            >
              Arugula and Walnut Salad
            </li>
          </a>
        </ul>
        <a
          class="add-link"
          href="/snacks/add"
        >
          Add a new item
        </a>
      </div>
    </div>
  </section>
</DocumentFragment>
`;
